cmd/commands/cobra.go

```diff
@@ -231,6 +231,39 @@ func NotBlank(flagName string) FlagsValidator {
 	}
 }
 
+// ExactlyOneOf returns a FlagsValidator that asserts that one and only one of the passed in flags is set.
+func ExactlyOneOf(flagNames ...string) FlagsValidator {
+	return FlagsValidationConjunction(
+		AtLeastOneOf(flagNames...),
+		AtMostOneOf(flagNames...),
+	)
+}
+
+// AllOf returns a FlagsValidator that asserts that all of the passed in flags are set.
+func AllOf(flagNames ...string) FlagsValidator {
```

Comment:
While there is technically nothing wrong with the implementation of this, I wonder if it's really necessary.
Cobra already has support for marking fields as required (and I think it does something extra wrt bash completion when they are marked).
The biggest "problem" I see with this is that, when I get 3 out of 4 flags correct, I'm greeted with an error message that sends me too much information, and it's a bit hard to see immediately what I did wrong.

Obviously, if we drop this, we get the reverse problem: fail at one flag, correct it, get the next error message, etc...

Reply:
Changed to mark the relevant flags required, thanks. Deleted `AllOf`.

This works out well as only the missing ones are complained about. For example, if I specify `--images` and  `--output` but forget `--registry` and `--registry-user`, the following error message is produced:
```
Error: required flag(s) "registry", "registry-user" not set
```